lib/Projects/ProjectDataReader.php

```diff
@@ -30,26 +30,55 @@ class ProjectDataReader
     /** @var mixed[] */
     private $projectsData;
 
+    /** @var mixed[] */
+    private $projectIntegrationTypes;
+
     /**
      * @param mixed[] $projectsData
+     * @param mixed[] $projectIntegrationTypes
      */
-    public function __construct(string $projectsPath, array $projectsData)
-    {
-        $this->projectsPath = $projectsPath;
-        $this->projectsData = $projectsData;
+    public function __construct(
+        string $projectsPath,
+        array $projectsData,
+        array $projectIntegrationTypes
+    ) {
+        $this->projectsPath            = $projectsPath;
+        $this->projectsData            = $projectsData;
+        $this->projectIntegrationTypes = $projectIntegrationTypes;
     }
 
     /**
      * @return mixed[]
      */
     public function read(string $repositoryName) : array
     {
-        return array_replace(
+        $projectData = array_replace(
```

Comment:
Would it make sense to create a VO for the project data?

Reply:
This is the array of data used to populate that Project model/vo.

Reply:
I know, but it isn't unusual that a VO gets other VOs injected. Project data itself is now used by two other VOs. We can have invalid (well, theoretically, because there are array key checks) arrays that should fail earlier by using a VO and may reduce later checks with `if` or `??`.

I made this a comment instead of a change request, because it isn't currently important for an approval. I just see a point where we can start to introduce the suggested VO.

Reply:
I will look at making that improvement in another PR.